### DISCUSSION TITLE
About Behavior of  `unregisterJsModule`

### DISCUSSION THREAD

**Discussion by nosuke23** [pyodide/pyodide#2033] [Q&A] [answered]
I want to restrict access to global window in python, so I tried to `unregisterJsModule` . 
However, after that, I was able to access to `js` in python. Is this behavior correct?

```js
let pyodide = await loadPyodide({indexURL: "..."})
pyodide.unregisterJsModule("js")
pyodide.runPython(`
import js
js
`) // globalThis
```

**Comment by hoodmane** [comment, marked as answer]
Once Pyodide is done loading, a bunch of different modules have already imported from `js` and have references to it. You'd have to go through and delete them all. It's sort of too late to fix the problem.

Instead use:
```js
const jsglobals = {
  // whatever stuff you want to expose, Pyodide will break if it is missing certain things though. 
};
let pyodide = await loadPyodide({indexURL: "...", jsglobals })
```
